packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageManager.java

```diff
@@ -1,28 +1,73 @@
 package expo.modules.image;
 
-import android.view.View;
+import android.widget.ImageView;
 
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.RequestManager;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReadableMap;
 import com.facebook.react.uimanager.SimpleViewManager;
 import com.facebook.react.uimanager.ThemedReactContext;
+import com.facebook.react.uimanager.annotations.ReactProp;
 
-import androidx.appcompat.widget.AppCompatCheckBox;
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 
-public class ExpoImageManager extends SimpleViewManager<View> {
+public class ExpoImageManager extends SimpleViewManager<ImageView> {
   private static final String REACT_CLASS = "ExpoImage";
 
+  private static final String URI_KEY = "uri";
+
+  private RequestManager mRequestManager;
+
+  public ExpoImageManager(ReactApplicationContext applicationContext) {
+    mRequestManager = Glide.with(applicationContext);
+  }
+
   @NonNull
   @Override
   public String getName() {
     return REACT_CLASS;
   }
 
+  // Props setters
+
+  @ReactProp(name = "source")
+  public void setSource(ImageView view, @Nullable ReadableMap sourceMap) {
+    if (sourceMap == null || sourceMap.getString(URI_KEY) == null) {
+      clearImageView(view);
+      return;
+    }
+
+    mRequestManager
+        .load(sourceMap.getString(URI_KEY))
+        .into(view);
+  }
+
+  // View lifecycle
+
   @NonNull
   @Override
-  public View createViewInstance(@NonNull ThemedReactContext context) {
-    // TODO: Implement some actually useful functionality
-    AppCompatCheckBox cb = new AppCompatCheckBox(c);
-    cb.setChecked(true);
-    return cb;
+  public ImageView createViewInstance(@NonNull ThemedReactContext context) {
+    ImageView imageView = new ImageView(context);
+
+    // For now let's set scale type to FIT_XY
+    // to make behavior same on all platforms.
+    imageView.setScaleType(ImageView.ScaleType.FIT_XY);
+
+    return imageView;
+  }
+
+  @Override
+  public void onDropViewInstance(@NonNull ImageView view) {
+    clearImageView(view);
```

Comment:
Is it necessary to clear the image view here? What does that solve? I can imagine that this can cause some drawing operations to be performed or added to the drawing pipeline.
I would keep this to the point and only clear any pending requests.